app/models/manageiq/providers/kubernetes/container_manager/refresh_parser.rb

```diff
@@ -134,31 +134,31 @@ def parse_node(node)
       new_result.merge!(
         :type                       => 'ManageIQ::Providers::Kubernetes::ContainerManager::ContainerNode',
         :identity_infra             => node.spec.externalID,
-        :identity_machine           => node.status.nodeInfo.machineID,
-        :identity_system            => node.status.nodeInfo.systemUUID,
-        :container_runtime_version  => node.status.nodeInfo.containerRuntimeVersion,
-        :kubernetes_proxy_version   => node.status.nodeInfo.kubeProxyVersion,
-        :kubernetes_kubelet_version => node.status.nodeInfo.kubeletVersion,
+        :identity_machine           => node.status.try(:nodeInfo).try(:machineID),
+        :identity_system            => node.status.try(:nodeInfo).try(:systemUUID),
+        :container_runtime_version  => node.status.try(:nodeInfo).try(:containerRuntimeVersion),
+        :kubernetes_proxy_version   => node.status.try(:nodeInfo).try(:kubeProxyVersion),
+        :kubernetes_kubelet_version => node.status.try(:nodeInfo).try(:kubeletVersion),
         :labels                     => parse_labels(node),
         :lives_on_id                => nil,
         :lives_on_type              => nil
       )
 
-      node_memory = node.status.capacity.try(:memory)
+      node_memory = node.status.try(:capacity).try(:memory)
       node_memory &&= parse_iec_number(node_memory) / 1.megabyte
```

Comment:
@blomquisg if `node_memory` is `nil` line 148 is not evaluated, therefore no exceptions. E.g.:

    > foo_bar = nil
    > foo_bar &&= foo_bar / 2
    => nil